Delay colorpick eyedrop preview using a timer

giarve opened the pull request:
This avoids recalculating the color of the selected pixel every time the mouse is moved.

Fixes #25358.

Calinou commented on scene/gui/color_picker.cpp:
```diff
@@ -855,6 +860,12 @@ ColorPicker::ColorPicker() :
 	bt_add_preset->set_tooltip(TTR("Add current color as a preset."));
 	bt_add_preset->connect("pressed", callable_mp(this, &ColorPicker::_add_preset_pressed));
+
+	mouse_not_moving_timer = memnew(Timer);
```
Add a comment to explain, as this isn't obvious at a glance:

```suggestion
	// Debounce color picker eyedrop preview to prevent the colorpicker from taking a long time to pick a color.
	mouse_not_moving_timer = memnew(Timer);
```

Calinou reviewed (changes requested):
Should be good to merge after applying the suggestion and squashing commits.

giarve committed:
Delay colorpick eyedrop preview using a timer

This avoids recalculating the color of the selected pixel every time the mouse is moved.

Fixes #25358.

KoBeWi commented on scene/gui/color_picker.cpp:
```diff
@@ -675,22 +675,26 @@ void ColorPicker::_screen_input(const Ref<InputEvent> &p_event) {
 
-	Ref<InputEventMouseMotion> mev = p_event;
-	if (mev.is_valid()) {
+	mouse_not_moving_pos = p_event;
+	mouse_not_moving_timer->start();
```
I'd move the event `is_valid()` check here. There's no point in starting the timer if the event is invalid. Also with the current code, `get_global_position()` might be called on invalid event.

KoBeWi commented:
This is impossible to test due to #46334, so probably won't be merged until that is resolved (unless we merge without testing...).
